core/src/main/java/org/kiwix/kiwixmobile/core/main/MainMenu.kt

```diff
@@ -144,6 +147,12 @@ class MainMenu(
   private fun navigateToSearch(zimFileReader: ZimFileReader): Boolean {
     activity.startActivityForResult(
       activity.intent<SearchActivity> {
+        when (originOfRequest) {
+          "TAB_SWITCHER" -> putExtra(EXTRA_COME_FROM, "TAB_SWITCHER")
+          else -> {
+            putExtra(EXTRA_COME_FROM, "")
```

Comment:
When you see a pattern where based on conditionals you only change one parameter in a method call it is usually a sign of an overly broad scope, you can move the conditional into the method and make it more concise 
```
putExtra(EXTRA_COME_FROM, if (originOfRequest == "TAB_SWITCHER") "TAB_SWITCHER" else "")
```
or a more spaced out
```
putExtra(
  EXTRA_COME_FROM, 
  if (originOfRequest == "TAB_SWITCHER") "TAB_SWITCHER" 
  else ""
)
```

I'd also maybe try and simplify this, you have duplicated magic strings that should be a constant.

Seeing as you only have 2 possible options here OpenedFromTabView or OpenedFromWebView I'd be happy enough passing this along as a Boolean ie 
```
setBooleanExtra(OPENED_FROM_TAB_VIEW, isInTabView)
.
.
.
fun showTabSwitcherOptions() {
.
isInTabView = true
}
.
.
.
fun showWebViewOptions(urlIsValid: Boolean) {
.
isInTabView=false
}
```

Reply:
Done!! (the explanation helps a lot 😂 😂) (alredy tested)